src/server/worker/pipeline/service/service.go

```diff
@@ -85,6 +87,9 @@ func forEachJob(pachClient *client.APIClient, pipelineInfo *pps.PipelineInfo, lo
 	var cancel func()
 	var eg *errgroup.Group
 	return pachClient.SubscribeJob(pipelineInfo.Pipeline.Name, true, func(ji *pps.JobInfo) error {
+		if ji.State == pps.JobState_JOB_FINISHING {
```

Comment:
does this usually get hit if we just finished processing a job?

Reply:
I don't think it should generally, as we'd pick up the new job event before the finishing happens. But we do the same in the normal transform worker, and I felt like it was better to be defensive here, since there's nothing we'd want to do on a FINISHING job, anyway.